test/unit/math/rev/fun/singular_values_U_test.cpp

```diff
@@ -0,0 +1,47 @@
+#include <stan/math/prim.hpp>
+#include <stan/math/rev/fun/singular_values_U.hpp>
+#include <stan/math/rev/fun/sum.hpp>
+#include <stan/math/rev/fun/log.hpp>
+#include <gtest/gtest.h>
+#include <test/unit/util.hpp>
+#include <typeinfo>
+TEST(AgradRev, svd_U_gradient) {
```

Comment:
We don't actually need tests in rev. The mixed mode tests will cover this. We do need tests in prim for `svd_U` though.

The assumption of `expect_ad` is that the prim version returns accurate values. Once that is true, it can finite difference and check all the autodiff.

You'll need to implement the `prim` checks though. See how they're done for `eigenvectors_sym` or `singular_values`. You can just compute some reference values in R or something to compare things against.